dataland-frontend/src/components/general/SlideShow.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div
+    ref="slider"
+    role="list"
+    :class="slidesContainerClasses"
+    @pointerdown="dragStartCondition"
+    @touchstart="dragStartCondition"
+  >
+    <slot />
+  </div>
+  <div v-if="slideCount > 1" :class="arrowsContainerClasses">
+    <button @click="move(-1)" aria-label="Previous slide" :class="leftArrowClasses" />
+    <button @click="move(1)" aria-label="Next slide" :class="rightArrowClasses" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onUnmounted, ref, defineEmits, toRefs } from "vue";
+
+const props = withDefaults(
+  defineProps<{
+    slidesContainerClasses: string;
+    arrowsContainerClasses: string;
+    leftArrowClasses: string;
+    rightArrowClasses: string;
+    slideCount: number;
+    initialCenterSlide?: number;
+    scrollScreenWidthLimit: number;
+    slideWidth: number;
+  }>(),
+  {
+    initialCenterSlide: 0,
+    slideWidth: 440,
+  },
+);
+
+const { slideCount, initialCenterSlide, scrollScreenWidthLimit, slideWidth } = toRefs(props);
```

Comment:
Sorry I am not familiar with the `toRefs()` method yet.
Can someone pls explain to me why it is necessary here?

Google says that it is used to make these props reactive.
But I always assumed that props are reactive per default, meaning that if the value of a prop of a Vue component changes, its DOM will automatically update. Is this assumption wrong?

Reply:
Props are reactive by default, but they lose their reactivity if deconstructured. That's when toRef and toRefs come to help

Reply:
Ok thanks for the explanation 👍 

But now I am asking myself why the reactivity of these four props must be maintained.

For `initialCenterSlide` and `slideWidth` I can still imagine some cases, where their values change, e.g. they might be initialized with their default value at first (for some very short amount of time), and afterwards change to a value passed by the parent.

But `slideCount` and `scrollScreenWidthLimit` are required props, that won't change their value anymore.
Why is their reactivity required here?